src/Microsoft.AspNet.Mvc.Core/ActionContext.cs

```diff
@@ -13,6 +14,21 @@ namespace Microsoft.AspNet.Mvc
     public class ActionContext
     {
         /// <summary>
+        /// Creates a empty <see cref="ActionContext"/>.
+        /// </summary>
+        /// <remarks>
+        /// The default constructor should be used only when <see cref="ActionContext"/> needs to be directly
+        /// instantiated in user codes, such as for unit tests.
+        /// </remarks>
+        public ActionContext()
+        {
+            ActionDescriptor = new ActionDescriptor();
+            ModelState = new ModelStateDictionary();
+            HttpContext = new DefaultHttpContext();
+            RouteData = new RouteData();
```

Comment:
@yishaigalatzer and I had a discussion about this. This isn't what we discussed. 

Please do the following changes:

**only** `ModelState` should be initialized in this constructor
**only** `ModelState` should remain a get-only property

All other properties (`ActionDescriptor`, `HttpContext`, `RouteData`) should be left uninitialized by this constructor and be get/set properties.